core/src/commonMain/kotlin/com/powersync/db/PowerSyncDatabaseImpl.kt

```diff
@@ -247,4 +254,38 @@ internal class PowerSyncDatabaseImpl(
             }
         }
     }
+
+    override suspend fun disconnect() {
+        if(syncJob != null && uploadJob != null && syncJob!!.isActive && uploadJob!!.isActive) {
+            //Wait for job to finish and then cancel
+            syncJob?.cancelAndJoin()
+            uploadJob?.cancelAndJoin()
+            syncStream = null
+            currentStatus.update(connected = false, connecting = false)
+        }
+    }
+
+    override suspend fun disconnectAndClear(clearLocal: Boolean) {
+        disconnect()
+
+        this.writeTransaction {
+            execute("DELETE FROM ${PsInternalTable.OPLOG}")
+            execute("DELETE FROM ${PsInternalTable.CRUD}")
+            execute("DELETE FROM ${PsInternalTable.BUCKETS}")
+            execute("DELETE FROM ${PsInternalTable.UNTYPED}")
+
+            val tableGlob = if (clearLocal) "ps_data_*" else "ps_data__*"
```

Comment:
Here is an `enum` and it should be updated to list both local and all data

https://github.com/powersync-ja/powersync-kotlin/blob/main/core/src/commonMain/kotlin/com/powersync/db/internal/PsInternalDatabase.kt#L36

Nit: Rather use a ternary instead of an inline `if`

Reply:
Is an `if` statement not an expression in kotlin. I don't think kotlin supports ternary operators.